security/webauthn/src/main/java/io/quarkus/ts/security/webauthn/security/MyWebAuthnSetup.java

```diff
@@ -9,79 +8,52 @@
-        return User.findByUserName(authenticator.getUserName())
-                .flatMap(user -> {
-                    // new user
-                    if (user == null) {
-                        User newUser = new User();
-                        newUser.userName = authenticator.getUserName();
-                        WebAuthnCredential credential = new WebAuthnCredential(authenticator, newUser);
-                        return credential.persist()
-                                .flatMap(c -> newUser.persist())
-                                .onItem().ignore().andContinueWithNull();
-                    } else {
-
-                        // existing user
-                        user.webAuthnCredential.counter = authenticator.getCounter();
-                        return Uni.createFrom().nullItem();
-                    }
-                });
-    }
-
-    private static Uni<List<Authenticator>> toAuthenticators(List<WebAuthnCredential> dbs) {
-        // can't call combine/uni on empty list
-        if (dbs.isEmpty())
-            return Uni.createFrom().item(Collections.emptyList());
-        List<Uni<Authenticator>> ret = new ArrayList<>(dbs.size());
-        for (WebAuthnCredential db : dbs) {
-            ret.add(toAuthenticator(db));
-        }
-        return Uni.combine().all().unis(ret).with(f -> (List) f);
+    public Uni<Void> store(WebAuthnCredentialRecord credentialRecord) {
+        User newUser = new User();
+        newUser.userName = credentialRecord.getUserName();
+        WebAuthnCredential credential = new WebAuthnCredential(credentialRecord, newUser);
+        return credential.persist()
+                .flatMap(c -> newUser.persist())
+                .onItem().ignore().andContinueWithNull();
     }
 
-    private static Uni<Authenticator> toAuthenticator(WebAuthnCredential credential) {
-        return credential.fetch(credential.webAuthnx509Certificates)
-                .map(x5c -> {
-                    Authenticator ret = new Authenticator();
-                    ret.setAaguid(credential.aaguid);
-                    AttestationCertificates attestationCertificates = new AttestationCertificates();
-                    attestationCertificates.setAlg(credential.alg);
-                    ret.setAttestationCertificates(attestationCertificates);
-                    ret.setCounter(credential.counter);
-                    ret.setCredID(credential.credID);
-                    ret.setFmt(credential.fmt);
-                    ret.setPublicKey(credential.publicKey);
-                    ret.setType(credential.type);
-                    ret.setUserName(credential.userName);
-                    return ret;
-                });
+    @WithTransaction
+    @Override
+    public Uni<Void> update(String credentialId, long counter) {
+        return WebAuthnCredential.findByCredentialId(credentialId)
+                .invoke(credential -> {
+                    // this user is handled in the LoginResource endpoint manually
```

Comment:
LoginResource was deleted. A bit confused by this comment.

Reply:
Oversight, I'll remove this.